### PR TITLE
fix: handle malformed URIs in prerequests

### LINKED ISSUE #28521: URIError: URI malformed in Cypress 13.6.1
> We're also seeing this after upgrading to Cypress 13.6.1 from 13.6.0.  We're running in a container on the docker image cypress/base:18.16.0.  In a suite of about 400 tests that are running parallel on 12 of these containers about 1/3 of the containers  will crash with the following error message.  Downgrading from 13.6.1 to 13.6.0 fixed the issue.

```
(Cypress:1324): dconf-CRITICAL **: 16:14:59.417: unable to create directory '/home/node/.cache/dconf': Permission denied.  dconf will not work properly.



(Cypress:1324): dconf-CRITICAL **: 16:14:59.417: unable to create directory '/home/node/.cache/dconf': Permission denied.  dconf will not work properly.



(Cypress:1324): dconf-CRITICAL **: 16:14:59.417: unable to create directory '/home/node/.cache/dconf': Permission denied.  dconf will not work properly.

URI malformed

URIError: URI malformed

    at decodeURI (<anonymous>)

    at u.addPending (<embedded>:4454:27423)

    at A.addPendingBrowserPreRequest (<embedded>:4454:37977)

    at a.addPendingBrowserPreRequest (<embedded>:4454:38767)

    at Q.addBrowserPreRequest (<embedded>:4526:13193)

    at c.onBrowserPreRequest (<embedded>:4526:27079)

    at T.onNetworkRequestWillBeSent (<embedded>:4454:49453)

    at _.emit (node:events:525:35)

    at _._handleMessage (<embedded>:2466:580235)

    at A.<anonymous> (<embedded>:2466:579669)

    at A.emit (node:events:513:28)

    at g.U (<embedded>:2466:564853)

    at g.emit (node:events:513:28)

    at g.dataMessage (<embedded>:2466:545228)

    at g.getData (<embedded>:2466:544410)

    at g.startLoop (<embedded>:2466:541117)

    at g._write (<embedded>:2466:540472)

    at writeOrBuffer (node:internal/streams/writable:392:12)

    at _write (node:internal/streams/writable:333:10)

    at Writable.write (node:internal/streams/writable:337:10)

    at Socket.H (<embedded>:2466:565539)

    at Socket.emit (node:events:513:28)

    at addChunk (node:internal/streams/readable:324:12)

    at readableAddChunk (node:internal/streams/readable:297:9)

    at Readable.push (node:internal/streams/readable:234:10)

    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)

We detected that the electron tab running Cypress tests closed unexpectedly.
```

_Originally posted by @marshallsanders in https://github.com/cypress-io/cypress/issues/28468#issuecomment-1847750031_

### PR DESCRIPTION
**Pull request by mschile**
- Closes #28521

### Additional details

In https://github.com/cypress-io/cypress/pull/28427 we started decoding URIs during the prerequest logic to resolve some correlation timeouts. However, we are not correctly handling malformed URIs which will cause Cypress to crash. This change tries to decode the URI and if it fails, just falls back to using the URI as-is for correlation.

### How has the user experience changed?

Cypress no longer crashes when receiving a malformed URI.

### PR Tasks

- [x] Have tests been added/updated?
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by chrisbreiding — APPROVED** [review]

**Review by jennifer-shehane — APPROVED** [review]

File: .vscode/settings.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "json"
   ],
   "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true
+    "source.fixAll.eslint": "explicit"
```
Comment (ryanthemanuel) [inline .vscode/settings.json:22]: What's the reason for this?

Comment (mschile) [reply .vscode/settings.json:22]: VSCode automatically updated it after I updated to the newest version of VSCode.

File: packages/proxy/lib/http/util/prerequests.ts [hunk 2]
```diff
@@ -320,4 +320,14 @@ export class PreRequests {
     this.pendingRequests = new QueueMap<PendingRequest>()
     this.pendingUrlsWithoutPreRequests = new QueueMap<PendingUrlWithoutPreRequest>()
   }
+
+  private tryDecodeURI (url: string) {
```
Comment (ryanthemanuel) [inline packages/proxy/lib/http/util/prerequests.ts]: This could be a static method right? Or external to the class?

Comment (mschile) [reply packages/proxy/lib/http/util/prerequests.ts]: yep, moved it, [43456a6](https://github.com/cypress-io/cypress/pull/28522/commits/43456a68294f9da935cf76e68bd4c775afc9ca25)

**Review by ryanthemanuel — APPROVED** [review]

**Pull request merged**